lib/diagnostic.rb

```diff
@@ -13,28 +13,29 @@ def longest(enum)
   # enumerable supplied (i.e. apply the `+` operator)
   # Assume each entry implements a `+` method
   def sum(enum)
-    # Your code goes here
+    enum.reduce(:+)
   end
 
   # Question 3
   # Write ruby code that returns true if some of the entries
   # in the numbers enumerable are odd
   def some_odd(numbers)
-    # Your code goes here
+    numbers.odd?
   end
 
   # Question 4
   # Write ruby code that returns true if every entry in the
   # numbers enumerable are even
   def every_even(numbers)
-    # Your code goes here
+    array = enum.each_with_object([]) { numbers.even? ? true : false }
+    array.include?(false) ? false : true
   end
 
   # Question 5
   # Write ruby code that returns an array with all of the
   # entries from the words enumerable capitalized (i.e. 'foo' becomes 'Foo').
   def transform(words)
-    # Your code goes here
+    words.each_with_object([]) { words.capitalize! }
```

Comment:
No reference inside the block to the given word.
Also, why use `each_with_object` if you're just going to operate in place?
`words.each { |word| word.capitalize! }`